File: Problem140_queue-reconstruction-by-height.java
// Time Complexity: O(n^2)
// Space Complexity: O(n)
// Did this code successfully run on Leetcode : Yes
// Any problem you faced while coding this : No


// Your code here along with comments explaining your approach
class Solution {
    public int[][] reconstructQueue(int[][] people) {
        if(people == null) 
            return new int[0][0];
        List<int[]> q = new ArrayList<>();
        Arrays.sort(people, (a, b) -> {
            if(a[0] == b[0]) return a[1] - b[1];
            return b[0] - a[0];
        });
        
        for(int i = 0; i < people.length; i++) {
            int[] person = people[i];
            q.add(person[1], person);
        }
        return q.toArray(new int[0][]);
    }
}

File: Problem-140 Queue Reconstruction by Height.py
# 406. Queue Reconstruction by Height
# https://leetcode.com/problems/queue-reconstruction-by-height/

# Time Complexity: O(nlogn)
# Space Complexity: O(1)

class Solution:
    def reconstructQueue(self, people: List[List[int]]) -> List[List[int]]:
        people.sort(key = lambda x: (-x[0], x[1]))
        
        res = []
        
        for i in people:
            res.insert(i[1], i)
        return res